### DISCUSSION TITLE
Replace deprecated PyDbg library with a Python 3-based debugger

### DISCUSSION THREAD

**Discussion by hippo-pat-amus** [jtpereyda/boofuzz#646] [Ideas]
As the PyDbg-based DebuggerThreadPydbg procmon thread was orphaned last year when the project made the switch to full Python 3, the target-based process monitor component is limited to the DebuggerThreadSimple type unless running an older version of Python on the target machine.

I propose the project adopt the Vivisect framework's Vtrace library (https://github.com/vivisect/vivisect) in order to re-obtain the debugger functionality lost with PyDbg. I've already made the switch and have the requisite files available in my own repo (https://github.com/hippo-pat-amus/boofuzz_vtrace), and have tested the new DebuggerThreadVtrace type on both Windows and Linux target platforms.

**Comment by blachine** [comment]
This is a really solid addition to the overall code base.  Your point about using older python code base is noted, but moving to Python 3 solution on the target saves from managing/working with two different versions and equally, if not more important, opens up new opportunities to further leverage other Vtrace/Vivesect functionality into our overall workflow.
